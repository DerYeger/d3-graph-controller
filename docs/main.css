:root {
  --color-primary: #729B1B;
  --color-secondary: #AF8A1D;
  --text-on-primary: white;
  --text-on-secondary: black;
}

@media (prefers-color-scheme: dark) {
  :root {
    --c-bg: rgb(20,20,20) !important;
    --c-text: rgb(255,255,255,87%) !important;
    --c-text-light: #fdd888 !important;
    --c-brand: #fdd888 !important;
    --c-accent: #90C321 !important;
    --bg-gradient: 20,20,20 !important;
    --code-inline-bg-color: #282c34aa !important;
  }
}

* {
  font-family: "Readex Pro", ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont, sans-serif;
}

.graph {
  border-radius: 6px;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  height: 480px !important;
  overflow: hidden;
  margin: 0 !important;
}

.nav-bar-title .logo {
  height: 1.5em !important;
  margin-right: 0.5rem !important;
}
