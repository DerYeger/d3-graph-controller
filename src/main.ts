export * from 'src/config/alpha'
export * from 'src/config/callbacks'
export * from 'src/config/config'
export * from 'src/config/filter'
export * from 'src/config/forces'
export * from 'src/config/initial'
export * from 'src/config/marker'
export * from 'src/config/modifiers'
export * from 'src/config/position'
export * from 'src/config/zoom'
export * from 'src/controller'
export * from 'src/model/graph'
export * from 'src/model/link'
export * from 'src/model/node'
