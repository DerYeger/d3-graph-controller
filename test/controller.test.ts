/**
 * @vitest-environment jsdom
 */
import { describe, expect, it } from 'vitest'
import { defineGraphConfig, GraphController } from '@src/main'
import TestData from '@test/test-data'

describe('GraphController', () => {
  it('renders nodes', () => {
    expect(window).toBeDefined()

    const container = document.createElement('div')
    new GraphController(container, TestData.graph, TestData.config)

    expect(container.querySelectorAll('.node').length).toEqual(
      TestData.graph.nodes.length
    )

    expect(container).toMatchSnapshot()
  })

  it('renders links', () => {
    expect(window).toBeDefined()

    const container = document.createElement('div')
    new GraphController(container, TestData.graph, TestData.config)

    expect(container.querySelectorAll('.link').length).toEqual(
      TestData.graph.links.length
    )

    expect(container).toMatchSnapshot()
  })

  it('respect initial configuration', () => {
    expect(window).toBeDefined()

    const container = document.createElement('div')
    new GraphController(
      container,
      TestData.graph,
      defineGraphConfig<string>({ initial: { nodeTypeFilter: [] } })
    )

    expect(container.querySelectorAll('.node').length).toEqual(0)
    expect(container.querySelectorAll('.link').length).toEqual(0)
  })
})
