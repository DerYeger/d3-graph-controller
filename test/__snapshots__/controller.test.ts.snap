// Vitest Snapshot v1

exports[`GraphController > renders links 1`] = `
<div>
  <svg
    height="100%"
    style="cursor: grab;"
    width="100%"
  >
    <g>
      <g
        class="links"
      >
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
      </g>
      <g
        class="nodes"
      >
        <g>
          <circle
            aria-label="A"
            class="node"
            r="16"
            style="fill: lightgray;"
          />
          <text
            class="node__label"
            dy="0.33em"
            opacity="1"
            style="fill: black; font-size: 1rem; stroke: none;"
          >
            A
          </text>
        </g>
        <g>
          <circle
            aria-label="B"
            class="node"
            r="16"
            style="fill: lightgray;"
          />
          <text
            class="node__label"
            dy="0.33em"
            opacity="1"
            style="fill: black; font-size: 1rem; stroke: none;"
          >
            B
          </text>
        </g>
        <g>
          <circle
            aria-label="C"
            class="node"
            r="16"
            style="fill: lightgray;"
          />
          <text
            class="node__label"
            dy="0.33em"
            opacity="1"
            style="fill: black; font-size: 1rem; stroke: none;"
          >
            C
          </text>
        </g>
      </g>
      <defs>
        <marker
          id="link-arrow-gray"
          markerHeight="4"
          markerWidth="4"
          orient="auto"
          refX="2"
          refY="2"
          style="fill: gray;"
          viewBox="0,0,4,4"
        >
          <path
            d="M0,0L0,4L4,2"
          />
        </marker>
      </defs>
    </g>
  </svg>
</div>
`;

exports[`GraphController > renders nodes 1`] = `
<div>
  <svg
    height="100%"
    style="cursor: grab;"
    width="100%"
  >
    <g>
      <g
        class="links"
      >
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
        <g>
          <path
            class="link"
            style="marker-end: url(#link-arrow-gray); stroke: gray;"
          />
          <text
            class="link__label"
            opacity="0"
            style="fill: black;"
          />
        </g>
      </g>
      <g
        class="nodes"
      >
        <g>
          <circle
            aria-label="A"
            class="node"
            r="16"
            style="fill: lightgray;"
          />
          <text
            class="node__label"
            dy="0.33em"
            opacity="1"
            style="fill: black; font-size: 1rem; stroke: none;"
          >
            A
          </text>
        </g>
        <g>
          <circle
            aria-label="B"
            class="node"
            r="16"
            style="fill: lightgray;"
          />
          <text
            class="node__label"
            dy="0.33em"
            opacity="1"
            style="fill: black; font-size: 1rem; stroke: none;"
          >
            B
          </text>
        </g>
        <g>
          <circle
            aria-label="C"
            class="node"
            r="16"
            style="fill: lightgray;"
          />
          <text
            class="node__label"
            dy="0.33em"
            opacity="1"
            style="fill: black; font-size: 1rem; stroke: none;"
          >
            C
          </text>
        </g>
      </g>
      <defs>
        <marker
          id="link-arrow-gray"
          markerHeight="4"
          markerWidth="4"
          orient="auto"
          refX="2"
          refY="2"
          style="fill: gray;"
          viewBox="0,0,4,4"
        >
          <path
            d="M0,0L0,4L4,2"
          />
        </marker>
      </defs>
    </g>
  </svg>
</div>
`;
